packages/experimental/theme/src/mergeTheme.ts

```diff
@@ -0,0 +1,6 @@
+import { immutableMerge } from '@fluentui-react-native/immutable-merge';
+import { Theme, PartialTheme } from '@fluentui-react-native/theme-types';
+
+export function mergeTheme<TTheme extends Theme, TPartial extends PartialTheme>(base: TTheme, partial: TPartial): TTheme {
+  return immutableMerge(base, (partial as unknown) as TTheme);
```

Comment:
is the casts to `unknown` necesary - and does immutableMerge not accept partials (or maybe TS isn't smart enough to realize it's a partial)?

Reply:
TS isn't smart enough because a Partial<MyType> is just a mapped type where it adds ? to every key within. The theme partial type is built up from explicitly declared partial sub-types. You can build a deep partial but then it does weird things to objects like ColorRamps where you really should be specifying multiple elements.

Reply:
Ah right the deep partials are what you want